lib/backend/k8s/resources/customresource.go

```diff
@@ -125,55 +119,30 @@ func (c *customK8sResourceClient) Update(ctx context.Context, kvp *model.KVPair)
 	logContext.Debug("Update custom Kubernetes resource")
 
 	// Create storage for the updated resource.
-	resOut := reflect.New(c.k8sResourceType).Interface().(CustomK8sResource)
+	resOut := reflect.New(c.k8sResourceType).Interface().(Resource)
 
 	var updateError error
-	for i := 0; i < 5; i++ {
-		// If no revision was passed, get the object to use its latest Revision number.
-		// If a revision was passed, then we should just use that.
-		if len(kvp.Revision) == 0 {
-			logContext.Debug("Querying for resource version")
-			k, err := c.Get(ctx, kvp.Key, kvp.Revision)
-			if err != nil {
-				return nil, err
-			}
-
-			if len(k.Revision) != 0 {
-				kvp.Revision = k.Revision
-				logContext.Debugf("Set resource version to %s", kvp.Revision)
-			}
-		}
-
-		// Convert the KVPair to a K8s resource.
-		resIn, err := c.converter.FromKVPair(kvp)
-		if err != nil {
-			logContext.WithError(err).Info("Error updating resource")
-			return nil, err
-		}
+	// Convert the KVPair to a K8s resource.
+	resIn, err := c.converter.FromKVPair(kvp)
+	if err != nil {
+		logContext.WithError(err).Info("Error updating resource")
+		return nil, err
+	}
 
-		// Send the update request using the name.
-		name := resIn.GetObjectMeta().GetName()
-		logContext = logContext.WithField("Name", name)
-		logContext.Debug("Update resource by name")
-		updateError = c.restClient.Put().
-			Resource(c.resource).
-			Body(resIn).
-			Name(name).
-			Do().Into(resOut)
-		if updateError == nil {
-			// Success.
-			// Update the revision information from the response.
-			kvp.Revision = resOut.GetObjectMeta().GetResourceVersion()
-			return kvp, nil
-		} else if kerrors.IsConflict(updateError) && len(kvp.Revision) == 0 {
-			// We only want to retry if there was no Revision provided with
-			// the KVP AND there was a CAS error while updating.
-			logContext.WithError(updateError).Warnf("Update failed for %s, retrying", kvp.Key.String())
-			continue
-		} else {
-			// We don't retry for any other errors or if the revision number is provided.
-			break
-		}
+	// Send the update request using the name.
+	name := resIn.GetObjectMeta().GetName()
+	logContext = logContext.WithField("Name", name)
+	logContext.Debug("Update resource by name")
+	updateError = c.restClient.Put().
+		Resource(c.resource).
+		Body(resIn).
+		Name(name).
+		Do().Into(resOut)
+	if updateError == nil {
+		// Success.
```

Comment:
A few lines down: https://github.com/projectcalico/libcalico-go/pull/582/files#diff-054d1c999e6b6ec72d35a9f7122ab93dR151

This is all garbage code now, right? We don't need to do a `Get` after a failed Update?

Reply:
grrr - yes - looks like i removed some but not all of the bits.  Thanks!

Reply:
hmmm actually - I'm not sure now

the etcdv3 code does return the snapshot on a failed update - which I basically copied from the k8s api server ... so I have a feeling we may need it for integration with the aapis.  I'll check.